From b1f674e323a41b258e8a4d7ede4367a0a52432c7 Mon Sep 17 00:00:00 2001
From: "David F. Skoll" <dfs@roaringpenguin.com>
Date: Tue, 2 Apr 2002 13:11:00 +0000
Subject: [PATCH] Add "nic-xxx" hack to rp-pppoe plugin.

---
 pppd/plugins/rp-pppoe/plugin.c | 10 +++++++---
 1 file changed, 7 insertions(+), 3 deletions(-)

diff --git a/pppd/plugins/rp-pppoe/plugin.c b/pppd/plugins/rp-pppoe/plugin.c
index c7c8923..a478410 100644
--- a/pppd/plugins/rp-pppoe/plugin.c
+++ b/pppd/plugins/rp-pppoe/plugin.c
@@ -22,7 +22,7 @@
 ***********************************************************************/
 
 static char const RCSID[] =
-"$Id: plugin.c,v 1.6 2002/03/14 21:54:06 dfs Exp $";
+"$Id: plugin.c,v 1.7 2002/04/02 13:11:00 dfs Exp $";
 
 #define _GNU_SOURCE 1
 #include "pppoe.h"
@@ -275,9 +275,13 @@ PPPoEDevnameHook(char *cmd, char **argv, int doit)
     int fd;
     struct ifreq ifr;
 
-    /* Only do it if name is "ethXXX" */
+    /* Only do it if name is "ethXXX" or "nic-XXXX.  In latter case,
+       strip off the "nic-" */
     /* Thanks to Russ Couturier for this fix */
-    if (strlen(cmd) < 4 || strncmp(cmd, "eth", 3)) {
+    if (strlen(cmd) > 4 && !strncmp(cmd, "nic-", 4)) {
+	/* Strip off "nic-" */
+	cmd += 4;
+    } else if (strlen(cmd) < 4 || strncmp(cmd, "eth", 3)) {
 	if (OldDevnameHook) return OldDevnameHook(cmd, argv, doit);
 	return 0;
     }
-- 
2.20.1

